subsys/mgmt/hawkbit/hawkbit.c

```diff
@@ -285,11 +290,11 @@ static int hawkbit_settings_set(const char *name, size_t len, settings_read_cb r
 
 #ifdef CONFIG_HAWKBIT_SET_SETTINGS_RUNTIME
 	if (settings_name_steq(name, "server_addr", &next) && !next) {
-		if (len != sizeof(hb_cfg.server_addr)) {
+		if (len > sizeof(hb_cfg.server_addr)) {
 			return -EINVAL;
 		}
 
-		rc = read_cb(cb_arg, &hb_cfg.server_addr, sizeof(hb_cfg.server_addr));
+		rc = read_cb(cb_arg, &hb_cfg.server_addr, len);
```

Comment:
what do you think about removing the check at the beginning and using `MIN(len, sizeof(hb_cfg.server_addr))` as len in `read_cb()` and using `strnlen()` after it to check if the string was NULL terminated?

Reply:
this way we could use INET_ADDRSTRLEN instead of INET6_ADDRSTRLEN, when IPv6 is not activated

Reply:
Wouldn't you want to ensure the string is null terminated before calling `read_cb`?

Reply:
If it would bepossible yes, but only in read_cb the string will be read, we don't have it before

Reply:
ah okay I was unfamiliar with what read_cb was doing. If it is copying a string into the buffer then sure - the check should be after.